### PR TITLE
feat: Add main menu page

### PR DESCRIPTION
**Pull request by jimbali**
Most of this is copied from the original `digital-prison-services` app. I have converted any JavaScript files to TypeScript.

[MAP-497]

[MAP-497]: https://dsdmoj.atlassian.net/browse/MAP-497?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: common/dateHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import moment from 'moment'
```
Comment (GurnankCheema) [inline common/dateHelpers.ts:1]: I think moment is deprecated now. 
 I know other projects are using  https://date-fns.org/

File: server/tests/testUtils.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import UserService from '../services/userService'
+
+export class MockUserService extends UserService {
+  constructor() {
+    super(undefined)
```
Comment (GurnankCheema) [inline server/tests/testUtils.ts:5]: does it need undefined here?

File: server/utils.ts [hunk 3]
```diff
@@ -0,0 +1,388 @@
+import moment from 'moment'
+import { DATE_TIME_FORMAT_SPEC } from '../common/dateHelpers'
+
```
Comment (GurnankCheema) [inline server/utils.ts:3]: maybe switch to a different date library like date-fns

File: server/utils.ts [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+ * correctly (i.e. each part in a double-barreled is converted to proper case).
+ * @param name name to be converted.
+ * @returns name converted to proper case.
+ */
+export const properCaseName = (name: string): string => (isBlank(name) ? '' : name.split('-').map(capitalize).join('-'))
+
+export const formatName = (firstName: string, lastName: string): string =>
+  [properCaseName(firstName), properCaseName(lastName)].filter(Boolean).join(' ')
+
+export const isViewableFlag = (code: string): boolean =>
```
Comment (GurnankCheema) [inline server/utils.ts:72]: not sure we will use this?

**Review by GurnankCheema — DISMISSED** [review]
LGTM

File: server/utils.ts (hunk 4 above)
Comment (jimbali) [reply server/utils.ts:72]: I think you're right, good spot

File: server/tests/testUtils.ts (hunk 2 above)
Comment (jimbali) [reply server/tests/testUtils.ts:5]: Yeah, if I remove it TypeScript complains with `Expected 1 arguments, but got 0.`

File: common/dateHelpers.ts (hunk 1 above)
Comment (jimbali) [reply common/dateHelpers.ts]: Turns out this file wasn't needed!

**Review by GurnankCheema — APPROVED** [review]
LGTM

**Pull request merged**